inference-engine/src/inference_engine/cnn_network_ngraph_impl.cpp

```diff
@@ -216,8 +216,11 @@ StatusCode CNNNetworkNGraphImpl::addOutput(const std::string& layerName, size_t
                 std::string outputName = layerName;
                 if (layer->outputs().size() != 1) {
                     outputName += "." + std::to_string(outputIndex);
+                } else if (outputIndex >= layer->outputs().size()) {
+                    return DescriptionBuffer(OUT_OF_BOUNDS, resp)
+                    << "port index " << outputIndex << " exceeds layer's outputs which is " << layer->outputs().size();
```

Comment:
can we add tests?

Reply:
sure, will add

Reply:
Added test for adding output with correct and incorrect ports